java/ql/lib/semmle/code/java/CompilationUnit.qll

```diff
@@ -31,5 +31,50 @@ class CompilationUnit extends Element, File {
    */
   Module getModule() { cumodule(this, result) }
 
+  /**
+   * Gets a type which is available by its simple name in this compilation unit.
+   * Reasons for this can be:
+   * - The type is declared in this compilation unit as top-level type
+   * - The type is imported
+   * - The type is declared in the same package as this compilation unit
+   * - The type is declared in the package `java.lang`
+   */
+  ClassOrInterface getATypeAvailableBySimpleName() {
+    // See "Shadowing", https://docs.oracle.com/javase/specs/jls/se17/html/jls-6.html#jls-6.4.1
+    // Note: Currently the logic below does not consider shadowing and might have multiple results
+    // with the same type name
+    result.(TopLevelType).getCompilationUnit() = this
+    or
+    exists(ImportStaticTypeMember importDecl |
+      importDecl.getCompilationUnit() = this and
+      result = importDecl.getATypeImport()
+    )
+    or
+    exists(ImportType importDecl |
+      importDecl.getCompilationUnit() = this and
+      result = importDecl.getImportedType()
+    )
+    or
+    exists(ImportStaticOnDemand importDecl |
+      importDecl.getCompilationUnit() = this and
+      result = importDecl.getATypeImport()
+    )
+    or
+    exists(ImportOnDemandFromType importDecl |
+      importDecl.getCompilationUnit() = this and
+      result = importDecl.getAnImport()
+    )
+    or
+    exists(ImportOnDemandFromPackage importDecl |
+      importDecl.getCompilationUnit() = this and
+      result = importDecl.getAnImport()
+    )
+    or
+    // From same package or java.lang, see https://docs.oracle.com/javase/specs/jls/se17/html/jls-7.html
```

Comment:
Add `CompilationUnit.getAnImport()` to make this much briefer?

Reply:
Do you mean as separate predicate: `getAnImport(): Import`?

Reply:
yes, so you don't need all the exists quantifiers but instead can do `this.getAnImport().(ImportStaticOnDemand).getAnImport()` for example

Reply:
I have simplified the code but decided against a separate predicate `getAnImport()` for know because I am not sure if it would be useful in any other situations.
I hope that is fine.